File: tiq-e/others/5/main.py
from time import perf_counter
from typing import List


class Solution:
    def isValid(self, s: str) -> bool:
        stack, match = [], {'{': '}', '[': ']', '(': ')'}
        for char in s:
            if char in match:
                stack.append(char)
            elif not stack or char != match[stack.pop()]:
                return False
        return not stack


x = "()[]{}"

start = perf_counter()
result = Solution().isValid(x)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/others/5/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn is_valid(s: String) -> bool {
        let mut stack = Vec::new();
        for char in s.chars() {
            match char {
                '{' => stack.push('}'),
                '(' => stack.push(')'),
                '[' => stack.push(']'),
                '}' | ')' | ']' if Some(char) != stack.pop() => return false,
                _ => (),
            }
        }
        return stack.is_empty();
    }
}

fn main() {
    let x: String = String::from("()[]{}");

    let start = Instant::now();
    let result = Solution::is_valid(x);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
